source/common/config/version_converter.cc

```diff
@@ -218,5 +219,30 @@ void VersionUtil::scrubHiddenEnvoyDeprecated(Protobuf::Message& message) {
   ProtobufMessage::traverseMutableMessage(proto_visitor, message, nullptr);
 }
 
+const Protobuf::MethodDescriptor&
+VersionUtil::getMethodDescriptorForVersion(const VersionedService* service,
+                                           envoy::config::core::v3::ApiVersion api_version,
+                                           bool use_alpha) {
+  std::string method_name;
+  switch (api_version) {
+  case envoy::config::core::v3::ApiVersion::AUTO:
+    FALLTHRU;
+  case envoy::config::core::v3::ApiVersion::V2:
+    method_name = fmt::format(service->methodNameTemplate(), use_alpha ? "v2alpha" : "v2");
```

Comment:
Is a simple template sufficient for all the services we want to cover? There will be some that might have moved package, but none that I can think of immediately.

Reply:
Let me try to implement all of the required services.

Reply:
@htuch I added the affected extensions as follow (and I think this PR is ready to review, while I'm still not sure about the structure and some naming for the newly introduced functions and classes):

Affected extensions:
- [x] filters/http/ext_authz
- [x] filters/network/ext_authz
- [x] filters/http/ratelimit
- [x] filters/network/ratelimit
- [x] access_loggers/grpc
- [x] stat_sinks/metrics_service

Notes for `tap`, `lrs`, `hds`, and `trace`:

1. `tap`
    - `TapSinkService` is not yet implemented.
    - `TapDiscoveryService` uses the`config.core.v3.ApiConfigSource`.
2. `lrs` uses the`config.core.v3.ApiConfigSource`.
3. `hds` uses the`config.core.v3.ApiConfigSource`.
4. `trace`
    - `StreamTraces` has no implemented client inside the code-base (implemented externally?)

I can check on `lrs`, `hds` if they are not ready (but it seems they have it) later.

If you want me to create multiple PRs for this, please let me know. Thank you!

Reply:
LRS/HDS have `ApiConfigSource` but I don't think the transport API version is meaningfully interpreted in them today. I think that's the main missing thing. Happy to follow whatever review structure you prefer, including single PR, since this is all fairly repetitive and easy to follow.